engines/wintermute/base/gfx/opengl/base_render_opengl3d_shader.h

```diff
@@ -0,0 +1,189 @@
+	void setLightParameters(int index, const Math::Vector3d &position, const Math::Vector3d &direction, const Math::Vector4d &diffuse, bool spotlight) override;
+
+	bool enableShadows() override;
+	bool disableShadows() override;
+	bool stencilSupported() override;
+
+	void dumpData(const char *filename) override {}
+	/**
+	 * Take a screenshot of the current screenstate
+	 *
+	 * @return a BaseImage containing the current screen-buffer.
+	 */
+	BaseImage *takeScreenshot() override;
+	bool saveScreenShot(const Common::String &filename, int sizeX = 0, int sizeY = 0) override;
+	void setWindowed(bool windowed) override;
+
+	/**
+	 * Fade a portion of the screen to a specific color
+	 *
+	 * @param r the red component to fade too.
+	 * @param g the green component to fade too.
+	 * @param b the blue component to fade too.
+	 * @param a the alpha component to fade too.
+	 * @param rect the portion of the screen to fade (if nullptr, the entire screen will be faded).
+	 */
+	void fadeToColor(byte r, byte g, byte b, byte a) override;
+
+	/**
+	 * Fill a portion of the screen with a specified color
+	 *
+	 * @param r the red component to fill with.
+	 * @param g the green component to fill with.
+	 * @param b the blue component to fill with.
+	 */
+	bool fill(byte r, byte g, byte b, Common::Rect *rect = nullptr) override;
+
+	bool setViewport(int left, int top, int right, int bottom) override;
+	bool drawLine(int x1, int y1, int x2, int y2, uint32 color) override;                // Unused outside indicator-display
+	bool drawRect(int x1, int y1, int x2, int y2, uint32 color, int width = 1) override; // Unused outside indicator-display
+
+	bool setProjection() override;
+	bool setProjection2D() override;
+	void resetModelViewTransform() override;
+	void setWorldTransform(const Math::Matrix4 &transform) override;
+
+	bool windowedBlt() override;
+
+	void onWindowChange() override;
+	bool initRenderer(int width, int height, bool windowed) override;
+	/**
+	 * Flip the backbuffer onto the screen-buffer
+	 * The screen will NOT be updated before calling this function.
+	 *
+	 * @return true if successfull, false on error.
+	 */
+	bool flip() override;
+	/**
+	 * Special flip for the indicator drawn during save/load
+	 * essentially, just copies the region defined by the _indicator-variables.
+	 */
+	bool indicatorFlip() override;
+	bool forcedFlip() override;
+	bool setup2D(bool force = false) override;
+	bool setup3D(Camera3D *camera, bool force = false) override;
+	bool setupLines() override;
+
+	/**
+	 * Get the name of the current renderer
+	 *
+	 * @return the name of the renderer.
+	 */
+	Common::String getName() const override {
+		return "OpenGL 3D renderer";
+	};
+	bool displayDebugInfo() override {
+		return STATUS_FAILED;
+	};
+	bool drawShaderQuad() override {
+		return STATUS_FAILED;
+	}
+
+	float getScaleRatioX() const override {
+		return 1.0f;
+	}
+	float getScaleRatioY() const override {
+		return 1.0f;
+	}
+
+	/**
```

Comment:
Might be best if overridden functions have their doxygen comments only in the class in which they're initially declared. Makes for less chance of documentation getting out of sync.

Reply:
Done